src/web/routes/application/enter-code/validate.js

```diff
@@ -1,9 +1,11 @@
+const { CONFIRMATION_CODE_SESSION_PROPERTY } = require('../common/constants')
 const { check } = require('express-validator')
 const { translateValidationMessage } = require('../common/translate-validation-message')
 
+const validateConfirmationCodeIsCorrect = () => (confirmationCode, { req }) => req.body.confirmationCode === req.session[CONFIRMATION_CODE_SESSION_PROPERTY]
+
 const validate = [
-  check('confirmationCode')
-    .not().isEmpty()
+  check('confirmationCode').custom(validateConfirmationCodeIsCorrect())
```

Comment:
The outer function can be removed here and this line would become:
```
check('confirmationCode').custom(validateConfirmationCodeIsCorrect)
```

Reply:
Removed outer function.